dashboard/common/utils.py

```diff
@@ -4,10 +4,21 @@
 from django.contrib.flatpages.models import FlatPage
 
 from dashboard.common.db_util import get_user_courses_info
+from dashboard.models import Course
 
 logger = logging.getLogger(__name__)
 
 
+def find_earliest_start_datetime_of_courses():
+    course_start_datetimes = []
+    for course in Course.objects.all():
+        course_start_datetimes.append(course.get_course_date_range().start)
+    logger.debug(course_start_datetimes)
+    earliest_start = sorted(course_start_datetimes)[0]
```

Comment:
This should *probably* do something if course_start_datetimes is still empty (like if there's no courses)

Though the cron won't run if there's not any courses and this is currently only called from the cron.

Reply:
addressed in the latest code.